Book/php7/zend_engine/zend_opcache.rst

```diff
@@ -1,2 +1,85 @@
 Zend OPCache
 ============
+
+While PHP is an interpreter, it does not interpret PHP source code directly. Instead, the source code is first parsed
+into an abstract syntax tree (AST), which is then compiled to OPcodes (and used to generate classes and other data
+structures), which are then passed to the Zend virtual machine (Zend VM) to actually be interpreted.
+
+As you can see, there are quite a few steps involved to run a single PHP file. Without an additional extension, PHP
+will perform all of these steps for every single loaded file in every single request. Since usually PHP files don't
+change between requests both the parsing and compilation steps will always yield the same result. OPcache is an
+extension that caches the OPcodes and data structures like classes between requests to improve performance. The AST is
+not used after the compilation step (with the exception of constant expressions) and thus does not need to be cached.
+
+Normally, all the memory allocated during the processing of a request gets freed after the request has been processed.
+So to store the compiled OPcodes for future requests OPcache puts them in a shared memory segment (shm) that can not
+only be accessed by the same process after the request has finished but also by other processes handling other
+requests. This means that if you are using php-fpm with many children they will only store the OPcodes for each PHP
+file once in shm. 
+
+This comes with a significant restriction: The OPcodes and other data structures in shm must not be mutated by any
+child request (at least not unless these changes should also affect the other processes). If any changes to shm are
+made there is a locking mechanism to avoid data races.
+
+The classes and OParrays (arrays of OPcodes, i.e. functions) in shm are mostly immutable. There are currently three
```

Comment:
minor nit: I wouldn't use "three" here, because it may get outdated (but not updated), and it doesn't add real value anyway, so:
```suggestion
The classes and OParrays (arrays of OPcodes, i.e. functions) in shm are mostly immutable. There are the following
```